go/libraries/doltcore/sqle/tables.go

```diff
@@ -248,15 +256,19 @@ func (t *AlterableDoltTable) AddColumn(ctx *sql.Context, column *sql.Column, ord
 		nullable = alterschema.Null
 	}
 
-	var defVal types.Value
+	var defaultVal types.Value
 	if column.Default != nil {
-		defVal, err = col.TypeInfo.ConvertValueToNomsValue(column.Default)
+		defaultVal, err = col.TypeInfo.ConvertValueToNomsValue(column.Default)
 		if err != nil {
 			return err
 		}
 	}
 
-	updatedTable, err := alterschema.AddColumnToTable(ctx, table, col.Tag, col.Name, col.TypeInfo, nullable, defVal, orderToOrder(order))
+	if err := alterschema.ValidateNewColumn(ctx, t.db.root, table, t.name, col.Tag, col.Name, col.TypeInfo, nullable, defaultVal); err != nil {
```

Comment:
See prev comment. AddColumn should do its own validation, unless there is some really compelling reason for this not to be the case.

Reply:
`ValidateNewColumn` now needs access to the `root_val` to check tag uniqueness.
I can reorganize this to plumb the root val into `AllColumnToTable`